### PR TITLE
Adjust paths in checkups to accommodate NixOS

### PR DESCRIPTION
**Pull request by RebeccaMahany**
A couple checkups refer to the `/usr/local/kolide-k2` directory, which will not exist on NixOS. This PR adjusts paths where necessary to find the appropriate location on NixOS.

<details>

<summary>A successful `launcher doctor` run</summary>

```
$ sudo ./launcher doctor --osqueryd_path /nix/store/9n4rqjklvv4a5gpwhnni18b4n042gidh-kolide-launcher-1.4.5/bin/osqueryd --hostname k2device.kolide.com
{"caller":"main.go:36","msg":"Launcher starting up","revision":"6de4a0ebd0e45ae49355369ff405332fe32d155a","severity":"info","ts":"2024-01-30T22:04:51.337826879Z","version":"1.4.5-14-g6de4a0e"}
 	Platform: platform: linux, architecture: amd64
 	Launcher Version: launcher_version 1.4.5-14-g6de4a0e
 	Host Info: hostname: nixos-2311-first.c.launcher-test-vms-156a-156a.internal, uptime: 11 days, 7 hours, 17 minutes, 18 seconds
✅	Process Report: found 3 kolide processes, at least one running as root or system
✅	Root directory contents: root directory (/var/kolide-k2/k2device.kolide.com/) contains 21 files
✅	Check communication with Kolide: successfully connected to device and control server
✅	Logs: debug.json is 544518 bytes, and was last modified at 2024-01-30 22:04:37.102442327 +0000 UTC
✅	Binary directory contents: binary directory (/nix/store/9n4rqjklvv4a5gpwhnni18b4n042gidh-kolide-launcher-1.4.5/bin) contains 2 files
✅	Enrollment Secret: claim for nababe
✅	Network Report: launcher can listen on local network
✅	Osquery: osqueryd version 5.11.0
❌	Launcher Flags: failed to stat /etc/kolide-k2/launcher.flags: stat /etc/kolide-k2/launcher.flags: no such file or directory
? 	Gnome Extensions: no gnome users found
✅	Quarantine: no quarantine directories found
✅	System Time: system time is within 5.000000 minutes of server date header, delta = 0.027195 minutes
✅	DNS Resolution: successfully resolved 5/5 hosts
✅	TUF: Successfully gathered release version launcher/linux/amd64/launcher-1.4.5.tar.gz from https://tuf.kolide.com:443/repository/targets.json
✅	Osquery Conflicts: No notable directories were detected
⚠️	Server Data: no server_data store in knapsack
✅	Osquery Data: hardware_vendor: Google, os_build: 23.11.20240117.8bf65f1, os_name: NixOS, os_version: 23.11 (Tapir), hardware_model: Google Compute Engine, hardware_serial: GoogleCloud-DD400E1384144374F694A68747581BD1, hardware_uuid: dd400e13-8414-4374-f694-a68747581bd1

Checkups with warnings:
	* Server Data


Checkups with failures:
	* Launcher Flags
```

</details>

### CODE REVIEW

File: ee/debug/checkups/binary_directory.go [hunk 1]
```diff
@@ -68,10 +71,46 @@ func getBinDir() string {
 	case "darwin":
 		return "/usr/local/kolide-k2"
 	case "linux":
+		if allowedcmd.IsNixOS() {
```
Comment (directionless) [inline ee/debug/checkups/binary_directory.go:74]:
```suggestion
	case "darwin", "linux":
		if allowedcmd.IsNixOS() {
```

I think?

File: ee/debug/checkups/binary_directory.go [hunk 2]
```diff
@@ -68,10 +71,46 @@ func getBinDir() string {
...
 
 	return ""
 }
+
+// getBinDirOnNixOS returns the most likely binary directory for NixOS,
+// looking through the nix store to find it.
+func getBinDirOnNixOS() string {
+	// The binary directory on NixOS will look like, e.g.,:
+	// `/nix/store/w8k7h0s9v64gqj60as756gg81c788zkp-kolide-launcher-1.4.4-6-g33c6fd9/bin`
+	matches, err := filepath.Glob("/nix/store/*-kolide-launcher-*/bin")
```
Comment (directionless) [inline ee/debug/checkups/binary_directory.go:90]: Is `os.Executable() ` a viable pattern here? Maybe with `path.EvalSymlinks` ?

Comment (RebeccaMahany) [reply ee/debug/checkups/binary_directory.go:90]: I thought about that, but didn't know if `os.Executable` would be correct when launcher has auto-updated?

File: ee/debug/checkups/binary_directory.go (hunk 1 above)
Comment (RebeccaMahany) [reply ee/debug/checkups/binary_directory.go:74]: I don't think so -- `IsNixOS` is an OS check, not if the package manager is in use. I kept it pretty narrowly scoped to start.

I am not sure how much of the changes I've made so far adequately support Nix on Darwin right now -- could be fine, but I haven't tested it at all yet.

File: ee/debug/checkups/binary_directory.go (hunk 2 above)
Comment (directionless) [reply ee/debug/checkups/binary_directory.go:90]: Good question. Seems easy to test.

Comment (directionless) [reply ee/debug/checkups/binary_directory.go:90]: i wrote myself a test script. Using `execwrapper.Exec` on macOS, `os.Executable` works as you'd expect:
```go
package main

import (
	"context"
	"flag"
	"fmt"
	"os"
	"path/filepath"

	"github.com/kolide/launcher/pkg/execwrapper"
)

func main() {
	fs := flag.NewFlagSet("forker", flag.ExitOnError)
	var (
		flChild = fs.String("child", "", "fork and call child (if present)")
	)

	if err := fs.Parse(os.Args[1:]); err != nil {
		fmt.Printf("Got error parsing: %s\n", err)
		os.Exit(1)
	}

	p, err := os.Executable()
	if err != nil {
		fmt.Printf("Got error calling os.Executable: %s\n", err)
		os.Exit(1)
	}

	preal, err := filepath.EvalSymlinks(p)
	if err != nil {
		fmt.Printf("Got error calling filepath.EvalSymlinks: %s\n", err)
		os.Exit(1)
	}

	fmt.Printf("Hi! I am:        %s\n", p)
	fmt.Printf("which is really: %s\n", preal)

	ctx := context.Background()

	if *flChild != "" {
		fmt.Printf("I am calling:    %s\n\n", p)

		if err := execwrapper.Exec(ctx, *flChild, []string{""}, os.Environ()); err != nil {
			fmt.Printf("Got error from exec: %s\n", err)
			os.Exit(1)
		}
	}
}
```

```shell
$ go build -o /tmp/f ./test-cmds/fork-os.exec/
$ go run ./test-cmds/fork-os.exec/ --child /tmp/f 
Hi! I am:        /var/folders/tk/p7rq1__x1jv4sdyj4cmy7w580000gn/T/go-build3289246093/b001/exe/fork-os.exec
which is really: /private/var/folders/tk/p7rq1__x1jv4sdyj4cmy7w580000gn/T/go-build3289246093/b001/exe/fork-os.exec
I am calling:    /var/folders/tk/p7rq1__x1jv4sdyj4cmy7w580000gn/T/go-build3289246093/b001/exe/fork-os.exec

Hi! I am:        /tmp/f
which is really: /private/tmp/f
```

Comment (RebeccaMahany) [reply ee/debug/checkups/binary_directory.go:90]: So, we don't want to use `os.Executable()`, then

**Review by James-Pickett — APPROVED** [review]

File: ee/debug/checkups/binary_directory.go (hunk 2 above)
Comment (directionless) [reply ee/debug/checkups/binary_directory.go:90]: Hrm. I guess that's true, and this is a checkup so we really want the store contents.

**Review by directionless — APPROVED** [review]

**Pull request merged**